x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/rerank/ElasticInferenceServiceRerankTaskSettings.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.elastic.rerank;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.TaskSettings;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+
+public class ElasticInferenceServiceRerankTaskSettings implements TaskSettings {
+
+    public static final String NAME = "elastic_rerank_task_settings";
+    public static final String TOP_N_DOCS_ONLY = "top_n";
+
+    public static final ElasticInferenceServiceRerankTaskSettings EMPTY_SETTINGS = new ElasticInferenceServiceRerankTaskSettings(
+        (Integer) null
+    );
+
+    public static ElasticInferenceServiceRerankTaskSettings fromMap(Map<String, Object> map) {
+        ValidationException validationException = new ValidationException();
+
+        if (map == null || map.isEmpty()) {
+            return EMPTY_SETTINGS;
+        }
+
+        Integer topNDocumentsOnly = extractOptionalPositiveInteger(
```

Comment:
I think we can omit this class. We've moved the common rerank parameters up to the root level of the request and they're passed in to the `infer()` call from the `InferenceAction` class. So I think we'll want the `ElasticInferenceService` to Override the `validateRerankParameters` from `SenderService` to ensure only top n is set.

Here's where that's being called by the `SenderService`: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/SenderService.java#L158

Reply:
Sorry, I think I might have pointed you in the wrong direction here. After thinking about this more, should we allow a user to set a value that is persisted (aka a default). Meaning if the top n isn't included in the request, do we want the EIS integration to retrieve a value from task settings?

I'm discussing this with the team as well. Other providers still allow these values to be placed in the task settings.

Reply:
That's correct, if the only things we were going to use the task settings for were top n and a return documents flag, then we do not need `ElasticInferenceServiceRerankTaskSettings`.